src/modules/mavlink/mavlink_messages.cpp

```diff
@@ -4204,11 +4202,12 @@ class MavlinkStreamDistanceSensor : public MavlinkStream
 				break;
 			}
 
-			msg.orientation = dist_sensor.orientation;
-			msg.min_distance = dist_sensor.min_distance * 100.0f; /* m to cm */
-			msg.max_distance = dist_sensor.max_distance * 100.0f; /* m to cm */
-			msg.current_distance = dist_sensor.current_distance * 100.0f; /* m to cm */
-			msg.covariance = dist_sensor.variance * 1e4f; // m^2 to cm^2
+			msg.current_distance = dist_sensor.current_distance * 1e3f; // m to cm
+			msg.id               = dist_sensor.id;
+			msg.max_distance     = dist_sensor.max_distance * 1e3f;     // m to cm
+			msg.min_distance     = dist_sensor.min_distance * 1e3f;     // m to cm
+			msg.orientation      = dist_sensor.orientation;
+			msg.covariance       = dist_sensor.variance * 1e4f;         // m^2 to cm^2
```

Comment:
@dagar, is worthwhile submitting changes for the `mavlink_distance_sensor_t` message type to use `variance` rather than `covariance`?  None of the distance sensors populate the `*.variance` field, but it seems like a footsnare allowing the fields to be divergent.

Reply:
@mcsauder that was already discussed and given the overall usage of the message, it's not possible to change the field naming.

Reply:
Thanks Nuno!